hibernate-reactive-core/src/main/java/org/hibernate/reactive/stage/impl/StageSessionFactoryImpl.java

```diff
@@ -29,23 +35,41 @@
 public class StageSessionFactoryImpl implements Stage.SessionFactory {
 
 	private final SessionFactoryImpl delegate;
+	private final VertxInstance vertxInstance;
+	private final Executor vertxExecutor;
 
 	public StageSessionFactoryImpl(SessionFactoryImpl delegate) {
 		this.delegate = delegate;
+		vertxInstance = delegate.getServiceRegistry().getService(VertxInstance.class);
+		vertxExecutor = runnable -> {
+			Context context = vertxInstance.getVertx().getOrCreateContext();
+			if ( Vertx.currentContext() == context ) {
+				runnable.run();
+			}
+			else {
+				context.runOnContext( x -> runnable.run() );
+			}
+		};
+	}
+
+	<T> CompletionStage<T> stage(Function<Void, CompletionStage<T>> stageSupplier) {
+		return voidFuture().thenComposeAsync(stageSupplier, vertxExecutor);
```

Comment:
Everyone, does this look right?

Reply:
I'm not sure if that's the way one is supposed to use the Async.
My expectation was that the executor should start a Thread. But it seems to work and also makes the examples pass when this branch is rebased on top of @Sanne PR: https://github.com/hibernate/hibernate-reactive/pull/490

LGTM

Reply:
Yea that line is definitely wrong: it's using the wrong formatting style :-)

Jokes aside, it seems reasonabe. I wouldn't expect any Thread to be started, as we should push such work onto the vert.x eventloop.

Reply:
I meant in general, when one passes an executor to the thenComposeAsync, not in this particular case.
But it was just a wrong assumption from my side.